src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -180,7 +180,8 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
                 return new FunctionMetadataProvider(
```

Comment:
Could we replace this instantiation with a single call to `ActivatorUtilities.CreateInstance<FunctionMetadataProvider>` to avoid the explicit service resolutions?

Reply:
Not all of the services are in the service container, but this is still doable if we only provide the extras:

```suggestion
                return ActivatorUtilities.CreateInstance<FunctionMetadataProvider>(
                    sp,
                    ActivatorUtilities.CreateInstance<WorkerFunctionMetadataProvider>(sp),
                    ActivatorUtilities.CreateInstance<HostFunctionMetadataProvider>(sp));
```

Will need to move the new parameter on `FunctionMetadataProvider` to earlier.